### PR TITLE
3.8.0

### PR DESCRIPTION
**Pull request by FaithBeam**
**Interesting Things**

* Add two tabs, Main and Extras
  * The Main tab contains the contents of the previous Main View
  * The Extras tab will contain new features to be added in the future
* Add the Unlock Cheats checkbox to the Extras tab
  * More information here: https://github.com/FaithBeam/Sims-1-Complete-Collection-Widescreen-Patcher/wiki/Extras
* Include extra information to the progress bar
  * It now shows how many items are left and what the current item is doing

**Boring Programmer Things**

* Use autofac instead of splat to register services
  * Splat is still used to resolve dependencies
* All of the utilities are now services
  * PatchUtility => ResolutionPatchService
  * UninstallUtility => UninstallService
  * WrapperUtility => WrapperService
* The ProgressService now reports when an uninstall occurs

This PR will open the gates for including other essential mods and patches to the patcher.

### CODE REVIEW

**Comment by FaithBeam** [comment]
Main Tab
image

Extras Tab
image

File: Sims1WidescreenPatcher.Core/Tabs/ExtrasTabViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    private CheckboxSelectionSnapshot _previousSnapshot;
+    private IAppState AppState { get; }
+
+    public ExtrasTabViewModel(CheckboxViewModelFactory creator, ICheatsService cheatsService, IAppState appState,
+        IProgressService progressService)
+    {
+        _cheatsService = cheatsService;
+        AppState = appState;
+        UnlockCheatsViewModel = (CheckboxViewModel)creator.Create("Unlock Cheats");
+        UnlockCheatsViewModel.ToolTipText = "Unlock cheats that were disabled in the release build the game";
```
Comment (Margen67) [inline Sims1WidescreenPatcher.Core/Tabs/ExtrasTabViewModel.cs:26]:
```suggestion
        UnlockCheatsViewModel.ToolTipText = "Unlock cheats that were disabled in the release build of the game.";
```

Comment (FaithBeam) [reply Sims1WidescreenPatcher.Core/Tabs/ExtrasTabViewModel.cs]: Thank you for the review and for sticking around :)

**Comment by FaithBeam** [comment]
Changed the notification theme to match the sims popup box. It looks pretty out of place on white background but I still think it's cute.

image

**Pull request merged**